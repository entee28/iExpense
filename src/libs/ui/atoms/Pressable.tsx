import React from 'react'
import {
  ViewStyle,
  TouchableOpacity,
  TouchableOpacityProps
} from 'react-native'

type Props = Pick<
  TouchableOpacityProps,
  'children' | 'disabled' | 'hitSlop' | 'onPress' | 'testID' | 'activeOpacity'
> &
  ViewStyle

export const Pressable = ({
  activeOpacity,
  children,
  disabled,
  hitSlop,
  onPress,
  testID,
  ...styles
}: Props) => {
  return (
    <TouchableOpacity
      activeOpacity={activeOpacity}
      children={children}
      testID={testID}
      disabled={disabled}
      hitSlop={hitSlop}
      onPress={onPress}
      style={styles}
    />
  )
}
