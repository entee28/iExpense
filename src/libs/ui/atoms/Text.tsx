import React, { useMemo } from 'react'
import { TextProps, TextStyle, Text as RNText } from 'react-native'
import colors from '../colors'

type Props = {
  children?: React.ReactNode
  textProps?: TextProps
  extraBold?: boolean
  bold?: boolean
  semiBold?: boolean
  medium?: boolean
  regular?: boolean
} & TextStyle

export enum FontFamily {
  extraBold = 'Nunito-ExtraBold',
  bold = 'Nunito-Bold',
  medium = 'Nunito-Medium',
  semiBold = 'Nunito-SemiBold',
  regular = 'Nunito-Regular'
}

export const Text = ({
  textProps,
  children,
  fontSize = 14,
  lineHeight = 22,
  color = colors.mono100,
  textAlign = 'auto',
  extraBold,
  bold,
  semiBold,
  medium,
  regular,
  ...styles
}: Props) => {
  const fontFamily = useMemo(() => {
    if (extraBold) {
      return FontFamily.extraBold
    }
    if (bold) {
      return FontFamily.bold
    }
    if (semiBold) {
      return FontFamily.semiBold
    }
    if (medium) {
      return FontFamily.medium
    }
    if (regular) {
      return FontFamily.regular
    }
    return FontFamily.regular
  }, [extraBold, bold, semiBold, medium, regular])

  return (
    <RNText
      {...textProps}
      style={[{ color, textAlign, lineHeight, fontSize, fontFamily }, styles]}>
      {children}
    </RNText>
  )
}
