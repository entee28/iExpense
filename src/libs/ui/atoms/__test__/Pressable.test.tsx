import { fireEvent, render } from '@testing-library/react-native'
import React from 'react'
import { Pressable } from '../Pressable'
import { Text } from '../Text'

it('renders children correctly', () => {
  const pressable = render(
    <Pressable>
      <Text color="red" testID="pressable_text">
        Hello
      </Text>
    </Pressable>
  )

  const text = pressable.getByTestId('pressable_text')
  expect(text.children).toContain('Hello')
  expect(text).toHaveStyle({
    color: 'red'
  })
})

it('correctly handles press event', () => {
  let pressed = false

  const press = () => {
    pressed = true
  }

  const pressable = render(<Pressable testID="pressable" onPress={press} />)

  fireEvent.press(pressable.getByTestId('pressable'))
  expect(pressed).toEqual(true)
})

it('renders styles correctly', () => {
  const pressable = render(
    <Pressable
      width={300}
      height={300}
      backgroundColor="red"
      testID="pressable"
    />
  )

  expect(pressable.getByTestId('pressable')).toHaveStyle({
    width: 300,
    height: 300,
    backgroundColor: 'red'
  })
})

it('renders correctly', () => {
  const snapshot = render(
    <Pressable alignItems="center" justifyContent="center" onPress={() => {}}>
      <Text color="red">Hello</Text>
    </Pressable>
  ).toJSON()

  expect(snapshot).toMatchSnapshot()
})
