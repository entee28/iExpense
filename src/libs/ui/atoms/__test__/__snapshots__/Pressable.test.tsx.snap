// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "red",
          "fontFamily": "Nunito-Regular",
          "fontSize": 14,
          "lineHeight": 22,
          "textAlign": "auto",
        },
        Object {},
      ]
    }
  >
    Hello
  </Text>
</View>
`;
